lib/spack/llnl/util/filesystem.py

```
@@ -296,14 +322,22 @@ def group_ids(uid=None):
     Returns:
         (list of int): gids of groups the user is a member of
     """
+    if is_windows:
+        tty.warn("Function is not supported on Windows")
+        return []
+
     if uid is None:
-        uid = os.getuid()
+        uid = getuid()
     user = pwd.getpwuid(uid).pw_name
     return [g.gr_gid for g in grp.getgrall() if user in g.gr_mem]
 
 
 def chgrp(path, group):
     """Implement the bash chgrp function on a single path"""
+    if is_windows:
+        tty.warn("Function is not supported on Windows")
+        return
```

IMO this should raise an exception, and we should avoid calling this function at all on windows (or invoking functions which transitively call this).